decorator.rb

```diff
@@ -0,0 +1,26 @@
+require './nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0, 9] if @nameable.correct_name.length > 10
+  end
+end
```

Comment:
- [ ] OPTIONAL:  The name has a maximum of 10 characters. If it's longer it should trim the word. If not it should return the name. So you can define it:
```ruby
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
end
```

Reply:
Ok, thank you. I will modify it.